src/middlewared/middlewared/plugins/system.py

```diff
@@ -370,8 +369,8 @@ async def dmidecode_info(self):
         dmidecode data is mostly static so cache the results
         """
 
-        if self.DMIDECODE_CACHE is None:
-            self.DMIDECODE_CACHE = {
+        if self.__dmidecode_cache is None:
```

Comment:
I meant, why not have DMIDECODE_CACHE = {
                'ecc-memory': None,
                'baseboard-manufacturer': None,
                'baseboard-product-name': None,
                'system-manufacturer': None,
                'system-product-name': None,
                'system-serial-number': None,
                'system-version': None,
            }

in there directly, or now

self.__dmidecode_cache = {
                'ecc-memory': None,
                'baseboard-manufacturer': None,
                'baseboard-product-name': None,
                'system-manufacturer': None,
                'system-product-name': None,
                'system-serial-number': None,
                'system-version': None,
            }

in init? Why do that later and check for None?

Reply:
Misunderstood on moving it to init, sorry. I've removed the extra None check.